File: May 2024 GFG SOLUTION/May-20.cpp
//{ Driver Code Starts
#include<bits/stdc++.h>
using namespace std;

// } Driver Code Ends


class Solution {
  public:
    long long int PowMod(long long int a, long long int b, long long int m) {
        if (b == 0) return 1; // Base case: a^0 is 1

        a %= m;
        long long half = PowMod(a, b / 2, m);

        if (b % 2 == 0) {
            return (half * half) % m;
        } else {
            return (((half * half) % m) * a) % m;
        }
    }
};


//{ Driver Code Starts.
int main(){
    int T;
    cin >> T;
    while(T--)
    {
    	long long int x, n, m;
    	cin >> x >> n >> m;
    	Solution ob;
    	long long int ans = ob.PowMod(x, n, m);
    	cout << ans <<"\n";
    }
	return 0;
}

// } Driver Code Ends up

File: May 2024 GFG SOLUTION/May-20.java
//{ Driver Code Starts
//Initial Template for Java

import java.util.*;
import java.lang.*;
import java.io.*;
class GFG
{
    public static void main(String[] args) throws IOException
    {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int T = Integer.parseInt(br.readLine().trim());
        while(T-->0)
        {
            String s = br.readLine();
            String[] S = s.split(" ");
            long x = Long.parseLong(S[0]);
            long n = Long.parseLong(S[1]);
            long m = Long.parseLong(S[2]);
            Solution ob = new Solution();
            long ans = ob.PowMod(x, n, m);
            System.out.println(ans);
        }
    }
}

// } Driver Code Ends


//User function Template for Java

class Solution {
    public long PowMod(long x, long n, long m) {
        if (n == 0) return 1; // Base case: x^0 is 1

        x %= m;
        long half = PowMod(x, n / 2, m);

        if (n % 2 == 0) {
            return (half * half) % m;
        } else {
            return (((half * half) % m) * x) % m;
        }
    }
}
